### PR TITLE
Always print full service name in bluechictl list-units

### LINKED ISSUE #767: bluechictl list-units output differs from systemctl
## Describe the bug

When running `bluechictl list-units` (or for the specific node), it outputs a table like below
```
NODE                |ID                                                         |   ACTIVE|      SUB
====================================================================================================
node-foo            |initrd-usr-fs.target                                       | inactive|     dead
node-foo            |auditd.service                                             |   active|  running
node-foo            |dev-nvme0n1p2.device                                       |   active|  plugged
node-foo            |systemd-machine-id-commit.service                          |   active|   exited
node-foo            |initrd-fs.target                                           | inactive|     dead
node-foo            |blockdev@dev-zram0.target                                  | inactive|     dead
node-foo            |dev-ttyS1.device                                           |   active|  plugged
node-foo            |tmp.mount                                                  |   active|  mounted
node-foo            |systemd-zram-setup@zram0.service                           |   active|   exited
node-foo            |sys-devices-platform-serial8250-serial8250:0-serial8250:0.2|   active|  plugged
node-foo            |systemd-pcrmachine.service                                 | inactive|     dead
```

The `ID`, however, is limited to 60 characters (see [here](https://github.com/eclipse-bluechi/bluechi/blob/main/src/client/method-list-units.c#L159)). systemctl outputs up to 150 characters here. 

## To Reproduce

Running `systemctl list-units` and `bluechictl list-units` and comparing the output. 

## Expected behavior

Either `bluechictl list-units` also outputs the ID dynamically up to 150 characters or statically this length.

### PR DESCRIPTION
**Pull request by mwperina**
Till now we have printed only first 60 characters of unit name in
bluechictl list-units. But this made testing list-units problematic on
real nodes, where some units has longer names, which caused duplicity
issues in integrations test. This patch dynamically enlarges column
sizes of list-units output, so bluechictl always prints full unit name.

Fixes: https://github.com/eclipse-bluechi/bluechi/issues/767
Signed-off-by: Martin Perina <mperina@redhat.com>

### CODE REVIEW

File: src/client/method-list-units.c [hunk 1]
```diff
@@ -153,18 +157,79 @@ void unit_list_unref(UnitList *unit_list) {
...
+                len += sep_len;                     // length of column separator string
+        }
+        va_end(args);
+
+        if (len > 0) {
+                // at least one column exists, remove column separator at the end
+                len -= sep_len;
+        }
+        return len;
+}
```
Comment (engelmi) [inline src/client/method-list-units.c]: This function is only used once and much of its complexity stems from being a variadic function. Therefore, I'd get rid of this function and do this in-place in `print_unit_list_simple`.

File: src/client/method-list-units.c [hunk 2]
```diff
@@ -153,18 +157,79 @@ void unit_list_unref(UnitList *unit_list) {
...
+        unsigned long max_node_len = strlen(node_title);
+        unsigned long max_id_len = strlen(id_title);
+        unsigned long max_active_len = strlen(active_title);
+        unsigned long max_sub_len = strlen(sub_title);
+
+        LIST_FOREACH(units, unit, unit_list->units) {
+                calc_max_col_len(unit->node, &max_node_len);
+                calc_max_col_len(unit->id, &max_id_len);
+                calc_max_col_len(unit->active_state, &max_active_len);
+                calc_max_col_len(unit->sub_state, &max_sub_len);
```
Comment (engelmi) [inline src/client/method-list-units.c]: I'd remove the `calc_max_col_len` and use `fmax` from `math.h` (https://linux.die.net/man/3/fmax) with a bit of type casting (should be no issue as `unsigned long` < `double`):
```C
max_node_len = fmax((double)max_node_len, (double)strlen(unit->node))
```

File: src/client/method-list-units.c [hunk 3]
```diff
@@ -153,18 +157,79 @@ void unit_list_unref(UnitList *unit_list) {
...
+                 max_node_len,
+                 col_sep,
+                 max_id_len,
+                 col_sep,
+                 max_active_len,
+                 col_sep,
+                 max_sub_len);
+
+        printf(fmt_str, "NODE", "ID", "ACTIVE", "SUB");
+        print_header_sep_line(
```
Comment (engelmi) [inline src/client/method-list-units.c]: You could use `memset` for this and remove the `print_header_sep_line`:
```C
// +1 to account for the terminating char \0
char sep_line[FMT_STR_MAX_LEN + 1];
// assuming line length is in line_length
memset(&sep_line, '=', sizeof(char) * line_length);
printf("%s", sep_line);
```

Comment (mwperina) [reply src/client/method-list-units.c]: Done

File: src/client/method-list-units.c (hunk 1 above)
Comment (mwperina) [reply src/client/method-list-units.c]: Done

File: src/client/method-list-units.c (hunk 2 above)
Comment (mwperina) [reply src/client/method-list-units.c]: Hmm, we need to link with math library for that ...

Comment (engelmi) [reply src/client/method-list-units.c]: Fair point - we don't use `math.h` anywhere yet. We can also keep the custom function, but then I'd implement a plain max function, e. g. like this:
```C
unsigned long max(unsigned long x, unsigned long y) {
   if (x > y) {
      return x;
   }
   return y;
} 
```
Both options work for me.

**Review by engelmi — APPROVED** [review]
LGTM

Regarding using math library or small custom implementation: 
Both options work are ok, I think, with a slight tendency to the custom implementation after seen the required change in meson.

File: src/client/method-list-units.c (hunk 2 above)
Comment (mwperina) [reply src/client/method-list-units.c]: Implemented specialized function as mentioned above instead of linking math library

**Pull request merged**